state/state.go

```diff
@@ -0,0 +1,81 @@
+package state
+
+import (
+	"time"
+
+	"github.com/camptocamp/terraboard/config"
+	"github.com/hashicorp/terraform/states/statefile"
+	log "github.com/sirupsen/logrus"
+)
+
+// LockInfo stores information on a State Lock
+type LockInfo struct {
+	ID        string
+	Operation string
+	Info      string
+	Who       string
+	Version   string
+	Created   *time.Time
+	Path      string
+}
+
+// Lock is a single State Lock
+type Lock struct {
+	LockID string
+	Info   string
+}
+
+// Version is a handler for state versions
+type Version struct {
+	ID           string
+	LastModified time.Time
+}
+
+// Provider is an interface for supported state providers
+type Provider interface {
+	GetLocks() (map[string]LockInfo, error)
+	GetVersions(string) ([]Version, error)
+	GetStates() ([]string, error)
+	GetState(string, string) (*statefile.File, error)
+}
+
+// Provider is a handler for the configured one
+var p Provider
```

Comment:
I'd rather about global variables if possible. Could you have `Configure` return the provider instead?

Reply:
Definitely, I refactored using locally scoped variables 👍